subprojects/plugins/src/integTest/groovy/org/gradle/api/tasks/JavaExecIntegrationTest.groovy

```diff
@@ -109,6 +136,36 @@ class JavaExecIntegrationTest extends AbstractIntegrationSpec {
         executedAndNotSkipped ":run"
     }
 
+    def 'arguments passed via command line matter in incremental check'() {
+        given:
+        buildFile << """
+            run.outputs.file "out.txt"
+        """
+
+        when:
+        run("run", "--args", "2")
+
+        then:
+        executedAndNotSkipped ":run"
+        assertOutputFileIs("2\n")
+
+        when:
+        run("run", "--args", "2")
+
+        then:
+        skipped ":run"
+
+        when:
+        file("out.txt").delete()
+
+        and:
+        run("run", "--args", "3")
```

Comment:
It will also rerun the task with `--args 2`, right?

Reply:
Yep, already fixed.